ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py

```diff
@@ -0,0 +1,339 @@
+    "mlmodel": mlmodel.MlModel,
+    "pipeline": pipeline.Pipeline,
+    "table": table.Table,
+    "topic": topic.Topic,
+    "user": user.User,
+    "testSuite": testSuite.TestSuite,
+}
+
+CACHED_EVENTS = []
+START_TS = str(
+    int(
+        (datetime.utcnow() - timedelta(days=0))
+        .replace(hour=0, minute=0, second=0, microsecond=0)
+        .timestamp()
+        * 1000
+    )
+)
+END_TS = str(
+    int(
+        (datetime.utcnow() - timedelta(days=0))
+        .replace(hour=23, minute=59, second=59, microsecond=999999)
+        .timestamp()
+        * 1000
+    )
+)
+
+
+class WebAnalyticEntityViewReportDataProcessor(DataProcessor):
+    """Processor class used as a bridge to refine the data"""
+
+    _data_processor_type = "WebAnalyticEntityViewReportData"
+
+    def fetch_data(self) -> Iterable[WebAnalyticEventData]:
+        if CACHED_EVENTS:
+            for event in CACHED_EVENTS:
+                yield event
+        else:
+            CACHED_EVENTS.extend(
+                self.metadata.list_entities(
+                    entity=WebAnalyticEventData,
+                    params={
+                        "startTs": START_TS,
+                        "endTs": END_TS,
+                        "eventType": "PageView",
+                    },
+                ).entities
+            )
+            for event in CACHED_EVENTS:
+                yield event
+
+    def _refine_entity_event(self) -> Generator[dict, WebAnalyticEventData, None]:
+        """Coroutine to process entity web analytic event
+
+        Yields:
+            Generator[dict, WebAnalyticEventData, None]: _description_
+        """
+        refined_data = {}
+        EntityObj = namedtuple("EntityObj", ["entity_type", "fqn"])
+
+        while True:
+            event = yield refined_data
+            split_url = [url for url in event.eventData.url.split("/") if url]  # type: ignore
+
+            if not split_url or split_url[0] not in ENTITIES:
+                continue
+
+            entity_obj = EntityObj(split_url[0], split_url[1])
+
+            if entity_obj.fqn not in refined_data:
+                entity = self.metadata.get_by_name(
+                    ENTITIES[entity_obj.entity_type],
+                    fqn=entity_obj.fqn,
+                    fields="*",
+                )
+
+                try:
+                    tags = (
+                        [tag.tagFQN.__root__ for tag in entity.tags]
+                        if entity.tags
+                        else None
+                    )
+                    entity_tier = get_entity_tier_from_tags(entity.tags)
+                    owner = entity.owner.name if entity.owner else None
+                    owner_id = str(entity.owner.id.__root__) if entity.owner else None
+                except AttributeError as exc:
+                    entity_tier = None
+                    tags = None
+                    owner = None
+                    owner_id = None
```

Comment:
Just to confirm, do we want to nullify all the values in case of an `AttributeError`? We can move them before the try block.